client/src/globals/UserContext.js

```diff
@@ -23,16 +39,35 @@ export const UserContextProvider = (props) => {
                 else return res.json()
             })
             .then(res => {
-                if (status === 200 && res.success === true) return setUserId(res.userId);
+                if (status === 200 && res.success === true) {
+                    setIsComplete(true)
+                    setIsSubscribed(res.isSubscribed)
+                    setUserEmail(res.userEmail)
+                    return setUserId(res.userId);
+                }
                 else throw Error("error");
             })
             .catch(err => {
+                setIsComplete(true)
                 setUserId("")
+                setUserEmail("")
+                setIsSubscribed(false)
             });
-    });
-    return(
+    },[]);
+
+    if (!isComplete) {
```

Comment:
@bugatti12345 In the case of when user refreshes or enters the url directly, I would need the state before rendering, so I copied what you did for private route.

Reply:
Did you test this? Cuz I've tried this in the morning, and it doesn't work well. That's why I tried to make a` PublicRoute`, but if this work, we don't really need a ` PublicRoute`

Reply:
And I think since the checkout and upgrade page will be in `PrivateRoute`, I think it would be set before render because the PrivateRoute will block the async until the set update

Reply:
hmm I know for sure it doesn't solve the login redirect issue. 

I don't think having the async blocker only in private route worked for me, thats why I added in the usercontext too.

Giving it some thought, i think private route only blocks it's own getcookie call right? It would still proceed regardless whether setUserId, setIsSubscribe have finished in usercontext. Also I've read somewhere that the state setters are async as well

Reply:
That make sense. I'll push the public route soon